src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -220,14 +220,40 @@ private static ImmutableArray<ParameterName> GenerateNames(IList<string> reserve
             // instead.
             var info = semanticModel.GetTypeInfo(expression, cancellationToken);
 
-            // If we can't determine the type, then fallback to some placeholders.
             var type = info.Type;
             var pluralize = Pluralize(semanticModel, type);
 
+            // We may be able to use the type's arguments to generate a name.
+            if (TryGenerateNameFromTypeArgument(type, capitalize, pluralize, out var typeArgumentParameterName))
+            {
+                return typeArgumentParameterName;
+            }
+
             var parameterName = type.CreateParameterName(capitalize);
```

Comment:
Could this change go in `CreateParameterName`? Looks like it already has cases for array types and pointer types, so generic types is maybe not too much of a stretch?

Reply:
Thanks for the review! I just updated the code to put the change instead in the core Workspaces layer so that it can be used by both C# and VB.